### ISSUE TITLE
Regression in PGP verification of decrypted data since #1179

### ISSUE THREAD

**Issue by papegaaij** [bcgit/bc-java#1251]
After upgrading to 1.72 our PGP code started giving incorrect signature validation errors when decrypting files. The exact same error can be reproduced using the `KeyBasedFileProcessor` example with the following command:
```
org.bouncycastle.openpgp.examples.KeyBasedFileProcessor -d encrypted.pgp pgpkey.asc tkhbackup
```
The expected output is `message integrity check passed`, but since commit 0132e856dddf759eae90a0e08cc8bca0acaa7705 #1179 , it gives `message failed integrity check`.

I've attached the private key and encrypted file used for this example. Both are dummy files generated during our integration test and can be used for any purpose you see fit.
[example.tar.gz](https://github.com/bcgit/bc-java/files/9768193/example.tar.gz)

**Comment by dghgit** [comment]
Thanks, that was a big help. Try https://www.bouncycastle.org/betas 173b04 or later. You just need the PGP jar. Let me know if it checks out and we will do a 172.1 release for PGP. Can't leave something like this floating around.

**Comment by dghgit** [comment]
Okay 172.1 for the BC PG API is now on https://www.bouncycastle.org/latest_releases.html

Give it a whirl, assuming it checks out with everyone I'll publish it to Maven Central

**Comment by papegaaij** [comment]
Thanks for the notice. I've tested 1.72.1 and can confirm that the issue is fixed for us.

**Comment by margussipria** [comment]
We had this problem with our application, giving error while validating signature:
```
[info]   Cause: java.io.IOException: unknown object in stream: -119
[info]   at org.bouncycastle.openpgp.PGPObjectFactory.nextObject(Unknown Source)
[info]   at org.bouncycastle.openpgp.PGPObjectFactory$1.getObject(Unknown Source)
[info]   at org.bouncycastle.openpgp.PGPObjectFactory$1.hasNext(Unknown Source)
[info]   at scala.collection.convert.JavaCollectionWrappers$JIteratorWrapper.hasNext(JavaCollectionWrappers.scala:41)
```

I can confirm that by changing to use `bcpg-jdk18on-173b04.jar` this issue is fixed.

**Comment by dghgit** [comment]
fixed in 1.73.
